src/main/scala/io/iohk/ethereum/blockchain/sync/regular/BlockBroadcast.scala

```diff
@@ -45,10 +45,12 @@ class BlockBroadcast(val etcPeerManager: ActorRef) {
 
   private def broadcastNewBlock(blockToBroadcast: BlockToBroadcast, peers: Map[PeerId, PeerWithInfo]): Unit =
     obtainRandomPeerSubset(peers.values.map(_.peer).toSet).foreach { peer =>
-      val message: MessageSerializable =
-        if (peers(peer.id).peerInfo.remoteStatus.protocolVersion.toByte == ProtocolVersions.ETC64.version)
-          blockToBroadcast.as64
-        else blockToBroadcast.as63
+      val remoteStatus = peers(peer.id).peerInfo.remoteStatus
+
+      val message: MessageSerializable = remoteStatus.protocolFamily match {
+        case ETH => blockToBroadcast.as63
```

Comment:
would be so cool to have this fail if a new block type is introduced. just a thought, no great ideas.

Reply:
Not sure what you mean here

Reply:
thinking out loud:
```scala
  private def broadcastNewBlock(blockToBroadcast: BlockToBroadcast, peers: Map[PeerId, PeerWithInfo]): Unit =
    obtainRandomPeerSubset(peers.values.map(_.peer).toSet).foreach { peer =>
      val remoteStatus = peers(peer.id).peerInfo.remoteStatus

      val message: MessageSerializable = {
        val capOpt = Capabilities.of(remoteStatus.protocolFamily, remoteStatus.protocolVersion)
        capOpt match {
          case Capabilities.Eth63Capability => blockToBroadcast.as63
          case Capabilities.Eth64Capability => blockToBroadcast.as63
          case Capabilities.Etc64Capability => blockToBroadcast.as64
          case None                         => throw new RuntimeException("Unknown capability, cannot broadcast block")
        }
      }

...

    def of(pf: ProtocolFamily, v: Int): Option[Capability] =
      (pf, v) match {
        case (ProtocolFamily.ETH, 63) => Some(Eth63Capability)
        case (ProtocolFamily.ETH, 64) => Some(Eth64Capability)
        case (ProtocolFamily.ETC, 64) => Some(Etc64Capability)
        case _                        => None
      }
```

and an ADT out of the capabilities?